.goreleaser.yml

```diff
@@ -1,48 +1,98 @@
+# NOTICE: This file is written with the assumption that it will be used in parca-dev/cross-builder.
+# - There are paths in this file that are specific to parca-dev/cross-builder and Github Actions.
+# - Unfortunately, Goreleaser does not support templating environment variables per build config.
+project_name: parca-agent
 before:
   hooks:
     - go mod tidy -compat=1.17
 env:
-  - CGO_CFLAGS=-I{{ .Env.BPF_OUT }}/libbpf/usr/include
-  - CGO_LDFLAGS={{ .Env.BPF_OUT }}/libbpf/libbpf.a
+  - CGO_ENABLED=1
+  - LDFLAGS=-fuse-ld=lld
+  - PKG_CONFIG=pkg-config
+  - CMD_CC=clang
+  - CC=clang
 builds:
+  # NOTICE: This build config assumes that the build runs on a x86_64 host.
   - main: ./cmd/parca-agent/
-    id: "parca-agent"
+    id: "parca-agent-amd64"
     binary: parca-agent
+    # https://goreleaser.com/customization/build/#reproducible-builds
+    mod_timestamp: '{{ .CommitTimestamp }}'
     env:
-      - CGO_ENABLED=1
-      - CC=clang
+      # Needed by CGO.
+      - CGO_CFLAGS=-target x86_64-pc-linux-gnu -I/__w/parca-agent/parca-agent/dist/libbpf/amd64/usr/include
+      - CGO_LDFLAGS=-target x86_64-pc-linux-gnu -lelf -lz /__w/parca-agent/parca-agent/dist/libbpf/amd64/libbpf.a
+      # Needed for compiling libbpf (defined in Make).
+      - CFLAGS=-g -O2 -Werror -Wall -std=gnu89 -target x86_64-pc-linux-gnu
```

Comment:
I didn't want to use CCP_FLAGS because it's semantically wrong as you said. We can pull some make magic for this one. Unfortunately, nothing can be done for `CGO_*` Go and Goreleaser doesn't allow :shrug: